### PR TITLE
Bugfix and improvements

### PR DESCRIPTION
**Pull request by babenek**
Windows style directory separator case did not resolved to exclude them.
File path are checked in lowercase
Unit tests and application tests were added

### CODE REVIEW

File: credsweeper/__main__.py [hunk 1]
```diff
@@ -107,14 +107,16 @@ def get_json_filenames(json_filename: str):
...
+    credsweeper = CredSweeper(
+        rule_path=args.rule_path,
+        ml_validation=args.ml_validation,
+        api_validation=args.api_validation,
+        json_filename=json_filename,
+        pool_count=args.jobs,
+        ml_batch_size=args.ml_batch_size,
+        ml_threshold=args.ml_threshold,
+        find_by_ext=args.find_by_ext,
+    )
```
Comment (meanrin) [inline credsweeper/__main__.py:119]: Please check this formatting 
It seems this conflicting our yapf config

https://github.com/google/yapf see `INDENT_CLOSING_BRACKETS` `DEDENT_CLOSING_BRACKETS`
dedent is false in our config https://github.com/Samsung/CredSweeper/blob/b960eb3304caa1652bd9e8922e1f1487e3dbb8ad/.style.yapf#L116

File: credsweeper/app.py [hunk 2]
```diff
@@ -29,16 +29,19 @@ class CredSweeper:
...
+            rule_path: Optional[str] = None,
+            ml_validation: bool = False,
+            api_validation: bool = False,
+            json_filename: Optional[str] = None,
+            use_filters: bool = True,
+            pool_count: int = 1,
+            ml_batch_size: Optional[int] = 16,
+            ml_threshold: Optional[Tuple[float, ThresholdPreset]] = None,
+            find_by_ext: bool = False,
+    ) -> None:
```
Comment (meanrin) [inline credsweeper/app.py:44]: Please see other comment about formatting

File: credsweeper/app.py [hunk 3]
```diff
@@ -147,10 +150,10 @@ def file_scan(self, file_provider: ContentProvider) -> List[Candidate]:
...
                              pattern=regex.compile(".*"))
                 ],
-                                      patterns=[regex.compile(".*")],
-                                      rule_name="Dummy candidate",
-                                      severity=Severity.INFO,
-                                      config=self.config)
+                    patterns=[regex.compile(".*")],
+                    rule_name="Dummy candidate",
+                    severity=Severity.INFO,
+                    config=self.config)
```
Comment (meanrin) [inline credsweeper/app.py:156]: Please see other comment about formatting

**Review by meanrin — DISMISSED** [review]

**Review by meanrin — APPROVED** [review]
LGTM, thanks for the fixes!

**Review by csh519 — APPROVED** [review]
Thank you for improvements!

LGTM 👍

**Pull request merged**